src/client.ts

```diff
@@ -3793,22 +3790,13 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         let totalKeyCount = 0;
         let keys: IMegolmSessionData[] = [];
 
-        const path = this.makeKeyBackupPath(targetRoomId!, targetSessionId!, backupInfo.version);
+        const path = this.makeKeyBackupPath(targetRoomId, targetSessionId, backupInfo.version);
 
-        const algorithm = await BackupManager.makeAlgorithm(backupInfo, async () => {
-            return privKey;
-        });
+        const backupDecryptor = await this.cryptoBackend!.getBackupDecryptor(backupInfo, privKey);
```

Comment:
How do we know that `this.cryptoBackend` is defined? Can we have a comment explaining?

Reply:
Oh wait, I see it on line 3786. In that case, I think we shouldn't need the `!`? If we do, please add a little comment saying that we bailed out earlier if we had no cryptoBackend

Reply:
Thx, removed